File: tests/TestBindings/Lua/TestInterface.lua
game = BWAPI.MockGame.new()
otherGame = BWAPI.MockGame.new()
unit1 = BWAPI.MockUnit.new()
unit2 = BWAPI.MockUnit.new()
unit3 = BWAPI.MockUnit.new()

function clientInfo()
  local info = game.clientInfo
  assert(type(info) == "table")
  local fn = function() end
  info["test"] = fn

  assert(info.test == fn)
  assert(game.clientInfo == info)
  assert(game.clientInfo.test == fn)

  local otherInfo = otherGame.clientInfo
  assert(type(otherInfo) == "table")
  assert(otherInfo.test == nil)

  assert(unit1.clientInfo ~= unit2.clientInfo)
  assert(unit2.clientInfo ~= unit3.clientInfo)
  assert(unit1.clientInfo ~= unit3.clientInfo)
  unit1.clientInfo.test = fn
  unit2.clientInfo.test = 5
  assert(unit1.clientInfo.test == fn)
  assert(unit2.clientInfo.test == 5)
end

File: tests/TestBindings/TestInterface.cpp
#include <catch.hpp>
#include <sol.hpp>
#include <BWAPI_Lua.h>
#include "BindingTestHelpers.h"
#include "MockGame.h"
#include "MockUnit.h"

using namespace BWAPI;
using namespace BWAPI_Lua;

TEST_CASE("interface bindings", "[interface]")
{
	sol::state lua;
	lua.open_libraries(sol::lib::base);
	auto BWAPI = lua.create_named_table("BWAPI");
	bindGame(BWAPI);
	bindUnit(BWAPI);
	TestBindings::bindMockGameImpl(BWAPI);
	TestBindings::bindMockUnitImpl(BWAPI);
	lua.script_file(TestBindings::getLuaTestPath(__FILE__));

	SECTION("client info")
	{
		sol::function test = lua["clientInfo"];
		REQUIRE_NOTHROW(test());
	}
}